src/bucket.ts

```diff
@@ -1789,6 +1798,100 @@ class Bucket extends ServiceObject {
     );
   }
 
+  enableLogging(
+    config: EnableLoggingOptions
+  ): Promise<SetBucketMetadataResponse>;
+  /**
```

Comment:
I believe a `callback` overload is needed here.

Reply:
Sorry, would you mind showing me what I should do? 😊 Currently I have it written for when just required configuration is passed in and the user is in promise mode, and then when the configuration is passed in as well as the optional callback.

Reply:
I believe the implementation version with optional `callback` (below) is not counted as an overload by TS. 
Then for TS users if callback is passed, the compiler will complain for `Expected 1 argument, but got 2`
![image](https://user-images.githubusercontent.com/7764119/66412016-b7515a00-e9c2-11e9-882b-84c793829bdf.png)

adding this overload will make the compiler happy
```ts
enableLogging(
    config: EnableLoggingOptions,
    callback: SetBucketMetadataCallback
  ): void;
```

Reply:
I haven't been able to reproduce that, but I'm sure it's something I'm doing wrong. I added it and pushed!

Reply:
You should be able to reproduce it by trying a `callback` mode system test.
```ts
    it('should enable logging on current bucket by default callback', done => {
      bucket.enableLogging({prefix: PREFIX}, (err, metadata) => {
        assert.ifError(err);
        assert.deepStrictEqual(metadata.logging, {
          logBucket: bucket.id,
          logObjectPrefix: PREFIX,
        });
        done();
      });
    });
```